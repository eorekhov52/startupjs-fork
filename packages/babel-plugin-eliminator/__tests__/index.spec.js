const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

const LONGER_EXAMPLE = /* js */`
  import usedByFoo from 'used-by-foo'
  import usedByDefault from 'used-by-default'
  import usedByBar from 'used-by-bar'

  const varInFoo = 'var-in-foo'
  const varInDefault = 'var-in-default'
  const varInBar = 'var-in-bar'

  export const foo = () => {
    return usedByFoo(varInFoo)
  }

  export default () => {
    return usedByDefault(varInDefault)
  }

  export function bar () {
    return usedByBar(varInBar)
  }
`

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {
    removeExports: ['preload', 'staticPreload']
  },
  tests: {
    'removes export function declaration. 1': /* js */`
      export function preload() {
        return {}
      }
    `,
    'removes export function declaration. 2': /* js */`
      export function staticPreload() {
        return {}
      }
    `,
    'removes references that are only used in server exports': /* js */`
      var readFile = require('fs').readFile
      import foo from 'foo'
      // This is kept since it's not used by ssr exports
      import { getStaticProps } from './'
      const a = 1
      function bar() {

      }
      var shouldHeep = 2
      var alsoShouldKeep =3
      console.log(alsoShouldKepp)
      export function preload() {
        return {a, bar, foo, readFile: readFile.toString()}
      }
    `,
    'removes export variable declaration': /* js */`
      export var preload= function() {
        return {}
      }
    `,
    'removes re-exports': /* js */`
      export {preload} from './'
    `,
    'keeps other exports': /* js */`
      export const a = 1
      export {preload} from './'
    `,
    'removes destructuring assignment (array)': /* js */`
      const [a,b,c] = d
      const e = d[0]
      export const preload = () => {
        console.log(a,b,c,e)
      }
    `,
    'longer example. Remove `foo`': {
      pluginOptions: {
        removeExports: ['foo']
      },
      code: LONGER_EXAMPLE
    },
    'longer example. Remove `foo`, `default`': {
      pluginOptions: {
        removeExports: ['foo', 'default']
      },
      code: LONGER_EXAMPLE
    },
    'throws error if both `removeExports` and `keepExports` are specified': {
      pluginOptions: {
        removeExports: ['foo'],
        keepExports: ['bar']
      },
      error: true,
      code: /* js */`
        export const foo = () => {}
        export function bar () {}
      `
    },
    'longer example. Keep `foo`': {
      pluginOptions: {
        keepExports: ['foo']
      },
      code: LONGER_EXAMPLE
    },
    'longer example. Keep `foo`, `default`': {
      pluginOptions: {
        keepExports: ['foo', 'default']
      },
      code: LONGER_EXAMPLE
    }
  }
})
