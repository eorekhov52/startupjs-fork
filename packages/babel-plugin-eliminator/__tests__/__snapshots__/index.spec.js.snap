// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-eliminator keeps other exports: keeps other exports 1`] = `

export const a = 1
export {preload} from './'

      ↓ ↓ ↓ ↓ ↓ ↓

export const a = 1;
export var preload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator longer example. Keep \`foo\`, \`default\`: longer example. Keep \`foo\`, \`default\` 1`] = `

import usedByFoo from 'used-by-foo'
import usedByDefault from 'used-by-default'
import usedByBar from 'used-by-bar'

const varInFoo = 'var-in-foo'
const varInDefault = 'var-in-default'
const varInBar = 'var-in-bar'

export const foo = () => {
  return usedByFoo(varInFoo)
}

export default () => {
  return usedByDefault(varInDefault)
}

export function bar () {
  return usedByBar(varInBar)
}

      ↓ ↓ ↓ ↓ ↓ ↓

import usedByFoo from "used-by-foo";
import usedByDefault from "used-by-default";
const varInFoo = "var-in-foo";
const varInDefault = "var-in-default";
export const foo = () => {
  return usedByFoo(varInFoo);
};
export default () => {
  return usedByDefault(varInDefault);
};
export var bar = 1;


`;

exports[`@startupjs/babel-plugin-eliminator longer example. Keep \`foo\`: longer example. Keep \`foo\` 1`] = `

import usedByFoo from 'used-by-foo'
import usedByDefault from 'used-by-default'
import usedByBar from 'used-by-bar'

const varInFoo = 'var-in-foo'
const varInDefault = 'var-in-default'
const varInBar = 'var-in-bar'

export const foo = () => {
  return usedByFoo(varInFoo)
}

export default () => {
  return usedByDefault(varInDefault)
}

export function bar () {
  return usedByBar(varInBar)
}

      ↓ ↓ ↓ ↓ ↓ ↓

import usedByFoo from "used-by-foo";
const varInFoo = "var-in-foo";
export const foo = () => {
  return usedByFoo(varInFoo);
};
export default 1;
export var bar = 1;


`;

exports[`@startupjs/babel-plugin-eliminator longer example. Remove \`foo\`, \`default\`: longer example. Remove \`foo\`, \`default\` 1`] = `

import usedByFoo from 'used-by-foo'
import usedByDefault from 'used-by-default'
import usedByBar from 'used-by-bar'

const varInFoo = 'var-in-foo'
const varInDefault = 'var-in-default'
const varInBar = 'var-in-bar'

export const foo = () => {
  return usedByFoo(varInFoo)
}

export default () => {
  return usedByDefault(varInDefault)
}

export function bar () {
  return usedByBar(varInBar)
}

      ↓ ↓ ↓ ↓ ↓ ↓

import usedByBar from "used-by-bar";
const varInBar = "var-in-bar";
export var foo = 1;
export default 1;
export function bar() {
  return usedByBar(varInBar);
}


`;

exports[`@startupjs/babel-plugin-eliminator longer example. Remove \`foo\`: longer example. Remove \`foo\` 1`] = `

import usedByFoo from 'used-by-foo'
import usedByDefault from 'used-by-default'
import usedByBar from 'used-by-bar'

const varInFoo = 'var-in-foo'
const varInDefault = 'var-in-default'
const varInBar = 'var-in-bar'

export const foo = () => {
  return usedByFoo(varInFoo)
}

export default () => {
  return usedByDefault(varInDefault)
}

export function bar () {
  return usedByBar(varInBar)
}

      ↓ ↓ ↓ ↓ ↓ ↓

import usedByDefault from "used-by-default";
import usedByBar from "used-by-bar";
const varInDefault = "var-in-default";
const varInBar = "var-in-bar";
export var foo = 1;
export default () => {
  return usedByDefault(varInDefault);
};
export function bar() {
  return usedByBar(varInBar);
}


`;

exports[`@startupjs/babel-plugin-eliminator removes destructuring assignment (array): removes destructuring assignment (array) 1`] = `

const [a,b,c] = d
const e = d[0]
export const preload = () => {
  console.log(a,b,c,e)
}

      ↓ ↓ ↓ ↓ ↓ ↓

export var preload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator removes export function declaration. 1: removes export function declaration. 1 1`] = `

export function preload() {
  return {}
}

      ↓ ↓ ↓ ↓ ↓ ↓

export var preload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator removes export function declaration. 2: removes export function declaration. 2 1`] = `

export function staticPreload() {
  return {}
}

      ↓ ↓ ↓ ↓ ↓ ↓

export var staticPreload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator removes export variable declaration: removes export variable declaration 1`] = `

export var preload= function() {
  return {}
}

      ↓ ↓ ↓ ↓ ↓ ↓

export var preload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator removes re-exports: removes re-exports 1`] = `

export {preload} from './'

      ↓ ↓ ↓ ↓ ↓ ↓

export var preload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator removes references that are only used in server exports: removes references that are only used in server exports 1`] = `

var readFile = require('fs').readFile
import foo from 'foo'
// This is kept since it's not used by ssr exports
import { getStaticProps } from './'
const a = 1
function bar() {

}
var shouldHeep = 2
var alsoShouldKeep =3
console.log(alsoShouldKepp)
export function preload() {
  return {a, bar, foo, readFile: readFile.toString()}
}

      ↓ ↓ ↓ ↓ ↓ ↓

// This is kept since it's not used by ssr exports
import { getStaticProps } from "./";
var shouldHeep = 2;
var alsoShouldKeep = 3;
console.log(alsoShouldKepp);
export var preload = 1;


`;

exports[`@startupjs/babel-plugin-eliminator throws error if both \`removeExports\` and \`keepExports\` are specified: throws error if both \`removeExports\` and \`keepExports\` are specified 1`] = `

export const foo = () => {}
export function bar () {}

      ↓ ↓ ↓ ↓ ↓ ↓

Error: unknown file: You cannot specify both \`removeExports\` and \`keepExports\`

`;
